src/resources/detectors/AwsEcsDetector.ts

```diff
@@ -0,0 +1,51 @@
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { Detector, Resource, ResourceDetectionConfig } from '@opentelemetry/resources';
+import { getUri, logger } from '../../utils';
+
+/**
+ * AwsEcsDetector detects the resources related with AWS ECS (EC2 and Fargate).
+ */
+export class AwsEcsDetector implements Detector {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async detect(_config?: ResourceDetectionConfig): Promise<Resource> {
+    const metadataUriV4 = process.env['ECS_CONTAINER_METADATA_URI_V4'];
+    const metadataUri = process.env['ECS_CONTAINER_METADATA_URI'];
+
+    if (!metadataUriV4 && !metadataUri) {
+      logger.debug('AwsEcsDetector failed: Process is not on ECS');
+      return Promise.resolve(Resource.EMPTY);
+    }
+
+    // Returns https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html#task-metadata-endpoint-v4-response
+    return Promise.all([
+      getUri(metadataUriV4 || metadataUri),
+      getUri(`${metadataUriV4 || metadataUri}/task`),
+    ])
+      .then((responses) => {
```

Comment:
Maybe we need to wrap this function in try-catch?

Reply:
The detector mechanism takes care of it